scheduler-driver/src/main/kotlin/gov/nasa/jpl/aerie/scheduler/plan/InMemoryEditablePlan.kt

```diff
@@ -55,16 +92,33 @@ data class InMemoryEditablePlan(
     uncommittedChanges.add(Edit.Create(resolved))
     resolved.validateArguments(lookupActivityType)
     plan.add(resolved.toSchedulingActivity(lookupActivityType, true))
+
+    for (simResults in simResultsUpToDate) {
+      simResults.get()?.stale = true
+    }
+    // create a new list instead of `.clear` because commit objects have the same reference
+    simResultsUpToDate = mutableSetOf()
+
     return id
   }
 
   override fun commit() {
+    // Early return if there are no changes. This prevents multiple commits from sharing ownership of the set,
+    // because new sets are only created when edits are made.
+    // Probably unnecessary, but shared ownership freaks me out enough already.
+    if (uncommittedChanges.isEmpty()) return
+
     val committedEdits = uncommittedChanges
     uncommittedChanges = mutableListOf()
-    commits.add(Commit(committedEdits))
+
+    // Create a commit that shares ownership of the simResults set.
+    commits.add(Commit(committedEdits, simResultsUpToDate))
```

Comment:
I suppose there's no good reason anymore. It's leftover from an earlier version of this class. I think I can rework it pretty easily.

Reply:
Ok I've reworked it to flatten the committed changes into one list of Edits instead of keeping a list of Commits. I'm still using the `Commit` object though just to keep the committed changes and up-to-date sim results for that commit associated together; but there's now only one Commit object at a time.